packages/cadl-ranch-api/src/request-validations.ts

```diff
@@ -34,6 +36,38 @@ export const validateBodyEquals = (request: RequestExt, expectedBody: unknown |
   }
 };
 
+export const validateXmlBodyEquals = (request: RequestExt, expectedBody: string): void => {
+  if (isBodyEmpty(request.rawBody)) {
+    throw new ValidationError(BODY_EMPTY_ERROR_MESSAGE, expectedBody, request.rawBody);
+  }
+
+  expectedBody = `<?xml version='1.0' encoding='UTF-8'?>` + expectedBody;
+
+  let actualParsedBody = "";
+  parseString(request.rawBody!, (err: Error | null, result: any): void => {
```

Comment:
I think this can be simplified to use the `parseStringPromise` version so you can do 

```
const actualParsedBody = await parseStringPromise(request.rawbody);
```

Reply:
I thought about that but wasn't sure if it was ok to change this (and callers) to async functions.

Reply:
I think its fair, specially as the code you have technically wouldn't work if the callback is actually called async(not sure when it happens)

Reply:
Yeah, changing it all to async causes the test server to return a 500 when the request body doesn't match the expected body.

Reply:
Hmm that's not true. I'm seeing 500 even with the sync version.